plugins/kubernetes-api/ts/client.ts

```diff
@@ -546,39 +546,56 @@ module KubernetesAPI {
     }
 
     private get _restUrl() {
+      var url = new URI();
+
       if (this.options.urlFunction && angular.isFunction(this.options.urlFunction)) {
-        var answer = this.options.urlFunction(this.options);
+        let answer = this.options.urlFunction(this.options);
         if (answer === null || !answer) {
           return null;
         }
-        return new URI(answer);
+        url = new URI(answer);
       } else {
-        return new URI(UrlHelpers.join(masterApiUrl(), this._path));
+        url = new URI(UrlHelpers.join(masterApiUrl(), this._path))
+      }
+
+      if (this.options.labelSelector !== null) {
+        url = url.addQuery({labelSelector: this.options.labelSelector});
       }
+
+      return url;
     }
 
     private get _wsUrl() {
+      var url = new URI();
+
       if (this.options.urlFunction && angular.isFunction(this.options.urlFunction)) {
-        var answer = this.options.urlFunction(this.options);
+        let answer = this.options.urlFunction(this.options);
         if (answer === null || !answer) {
           return null;
         }
-        return wsUrl(answer).query(<any>{ watch: true });
+        url = wsUrl(answer);
       } else {
-        var url = UrlHelpers.join(masterApiUrl(), this._path);
-        var location = Core.windowLocation();
-        if (location && url.indexOf("://") < 0) {
-          var hostname = location.hostname;
+        let urlStr = UrlHelpers.join(masterApiUrl(), this._path);
+        let location = Core.windowLocation();
```

Comment:
wondering why this construct is used here but not in the `_restUrl` method.

Reply:
I'm inclined to think that it is not used. My guess would be that the WebSocket API may not work well with relative path, while the polling client does. Anyway, we can leave it for further analysis.